RIOT-OS/RIOT
# boards: remove some of the non required variables export

### Contribution description

This PR cleans up some of the non required use of `export`. The variables are not used in external scripts but only by the build system itself (for configuration, etc), so there's no need to export them.




### Testing procedure

- A green Murdock
- **telosb and iotlab-m3**: verify that `make term` works (untested, I don't have this board...)
- **cc2650-launchpad** (and cc2650stk): verify that flashing a firmware works
- **arduino-due**: verify that flashing a firmware works with bossa
- **nz32-sc151**: verif that flashing with dfu works
- **esp32-wroom-32**: verify that `ENC28J60_PARAM_DEFINED` is only defined once (by adding `$(info DEBUG)` just before the variable is set 1 and checking the DEBUG message is only displayed once)
- **opencm904**: verify that flashing a firmware still works
- **stm32** (bluepill ?): verify that flashing a firmware with dfu (using blackmagic_dfu) still works




### Issues/PRs references

Related to #10850
